src/IO/WriteBufferFromS3.cpp

```diff
@@ -107,7 +107,14 @@ void WriteBufferFromS3::finalizeImpl()
 
 WriteBufferFromS3::~WriteBufferFromS3()
 {
-    finalizeImpl();
+    try
+    {
+        finalizeImpl();
+    }
+    catch (...)
+    {
+        tryLogCurrentException(__PRETTY_FUNCTION__);
```

Comment:
But this will hide write error, does not looks right...

Reply:
Let's merge this PR first and another PR later.